### PR TITLE
Supports bundling with `esbuild`

### LINKED ISSUE #42: Support bundling with `esbuild`
Bundling with [esbuild](https://esbuild.github.io/) currently fails on this package due to an issue on [line 15 of index.js](https://github.com/visionmedia/batch/blob/47a9b1d1f5cd8e4b32dbb74a52aa968cf9b58480/index.js#L15). It references a package called "emitter" that esbuild cannot resolve, thus esbuild fails.

### PR DESCRIPTION
**Pull request by r-token**
As discussed in #42, bundling with [esbuild](https://esbuild.github.io/) currently fails on this package due to an issue on [line 15 of index.js](https://github.com/visionmedia/batch/blob/47a9b1d1f5cd8e4b32dbb74a52aa968cf9b58480/index.js#L15). It references a package called "emitter" that esbuild cannot resolve, thus esbuild fails.

This PR simply wraps the `var Emitter = require('emitter');` line in a `try/catch` block so that even if esbuild cannot resolve it, it will not fail.

Closes #42

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,7 +12,11 @@ try {
   var EventEmitter = require('events').EventEmitter;
   if (!EventEmitter) throw new Error();
 } catch (err) {
-  var Emitter = require('emitter');
+  try {
+    var Emitter = require('emitter');
+  } catch (err) {
+    console.error('Unable to create an emitter:', err);
```
Comment (dougwilson) [inline index.js:18]: This should not proceed if neither package was able to be loaded.

**Review by dougwilson — CHANGES_REQUESTED** [review]

**Comment by dougwilson** [comment]
Please also add either a test for this or at least instructions here for how to reproduce the issue so I can create a test for it so this doesn't regress.

**Comment by r-token** [comment]
Sure. Here are some steps to recreate:

1) Install esbuild globally via `npm install esbuild -g`
2) From this project, run `esbuild index.js --bundle`

You should get an error message that says `✘ [ERROR] Could not resolve "emitter"` with the file and line number in question.

Here's [esbuild's getting started page](https://esbuild.github.io/getting-started/) if you need more info.

**Comment by dougwilson** [comment]
Sweet, thanks! I will push up a commit here soon that wraps that in to a test so we don't break it 👍 . Thank you so much and I'll get this merged and released either today or tomorrow :D

File: index.js (hunk 1 above)
Comment (r-token) [reply index.js]: I am now `throw`ing if no emitter can be created. If you'd prefer a different way to handle this, happy to work on it with you.

Comment (dougwilson) [reply index.js]: It's perfectly fine 👍

**Comment by r-token** [comment]
Hi @dougwilson, just checking in. Anything else you need from me on this?

**Comment by dougwilson** [comment]
Hi @r-token thanks for the ping and sorry for being silent on your PR. In the process of adding the test, I noticed this project was not running the CI. I had to message Travis CI to see why it wasn't using my credits and apparently I need to migrate the project. I am just getting that done so the tests will run, as it's a Travis CI step I'm adding to bundle the module and test the bundle.

**Comment by dougwilson** [comment]
Hi @r-token sorry, it looks like I am unable to push the tests to this PR. Usually this is enabled by default, sorry about that. I just need you to check the box on the right side here to enable edits from maintainers (https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

**Comment by r-token** [comment]
@dougwilson I don't have that checkbox, unfortunately. I think it's because I opened this PR from my organization's account instead of my personal account (https://github.com/isaacs/github/issues/1681).

If necessary, I can open a separate PR with the same changes from my personal account. If you have a different/better idea though, I'm all for it. Let me know what you think.

**Comment by dougwilson** [comment]
Ah, interesting, didn't know that. It's ni problem, I can still land everything. No need to reclone move changes and all that 👍. I will nake time tonight for you to get this merged and released.

**Comment by r-token** [comment]
Hi @dougwilson, what's the latest on this? You're not waiting on anything from me, are you?

**Pull request closed without merging**